unstructured/ingest/connector/pinecone.py

```diff
@@ -0,0 +1,143 @@
+import json
+import multiprocessing as mp
+import typing as t
+from dataclasses import dataclass
+
+from unstructured.ingest.error import DestinationConnectionError, WriteError
+from unstructured.ingest.interfaces import (
+    BaseConnectorConfig,
+    BaseDestinationConnector,
+    BaseIngestDoc,
+    BaseSessionHandle,
+    ConfigSessionHandleMixin,
+    IngestDocSessionHandleMixin,
+    WriteConfig,
+)
+from unstructured.ingest.logger import logger
+from unstructured.staging.base import flatten_dict
+from unstructured.utils import requires_dependencies
+
+
+@dataclass
+class PineconeSessionHandle(BaseSessionHandle):
+    service: "pinecone.Index"  # noqa: F821
+
+
+@DestinationConnectionError.wrap
+@requires_dependencies(["pinecone"], extras="pinecone")
+def create_pinecone_object(api_key, index_name, environment):
+    import pinecone
+
+    pinecone.init(api_key=api_key, environment=environment)
+    index = pinecone.Index(index_name)
+    logger.debug(f"Connected to index: {pinecone.describe_index(index_name)}")
+    return index
+
+
+@dataclass
+class SimplePineconeConfig(ConfigSessionHandleMixin, BaseConnectorConfig):
+    api_key: str
+    index_name: str
+    environment: str
+
+    def create_session_handle(self) -> PineconeSessionHandle:
+        service = create_pinecone_object(self.api_key, self.index_name, self.environment)
+        return PineconeSessionHandle(service=service)
+
+
+@dataclass
+class PineconeWriteConfig(IngestDocSessionHandleMixin, WriteConfig):
+    connector_config: SimplePineconeConfig
+    batch_size: int = 50
+    num_processes: int = 1
+
+
+@dataclass
+class PineconeDestinationConnector(BaseDestinationConnector):
+    write_config: PineconeWriteConfig
+    connector_config: SimplePineconeConfig
+
+    def initialize(self):
+        pass
+
+    def check_connection(self):
```

Comment:
Can we include anything in here to actually check the connection?

Reply:
Since we use session handles per process, and since there's no global initialization step for this connector (global meaning, 'above / before all processes') I intentionally haven't put anything in `initialize` (such as creating a session or checking connection). Please let me know if we'd like to do anything there.